### ISSUE TITLE
Gdk can't find seats

### ISSUE THREAD

**Issue by acrisci** [SirCmpwn/wlroots#103]
A basic gtk window test on the compositor example fails for me with:

> (gtk-test:16551): Gdk-CRITICAL **: gdk_seat_get_keyboard: assertion 'GDK_IS_SEAT (seat)' failed

Try this basic Gdk seat test:

```python
#!/usr/bin/env python3

import gi
gi.require_version('Gdk', '3.0')
from gi.repository import Gdk

DISPLAY='wayland-1'

display = Gdk.Display().open(DISPLAY)
seat = display.get_default_seat()
seats = display.list_seats()

print(seat)
print(seats)
```

You'll see `None` as `seat` and an empty list as `seats`.

This returns seats on both weston and sway.

**Comment by acrisci** [comment]
Gdk requires `wl_data_device_manager` in order to support `wl_seat`.

From `gdk/wayland/gdkdisplay-wayland.c`:

```c
  else if (strcmp (interface, "wl_seat") == 0)
    {
      static const char *required_device_manager_globals[] = {
        "wl_compositor",
        "wl_data_device_manager",
        NULL
      };

      if (has_required_globals (display_wayland,
                                required_device_manager_globals))
        _gdk_wayland_display_add_seat (display_wayland, id, version);
      else
        {
          SeatAddedClosure *closure;

          closure = g_new0 (SeatAddedClosure, 1);
          closure->base.handler = seat_added_closure_run;
          closure->base.required_globals = required_device_manager_globals;
          closure->id = id;
          closure->version = version;
          postpone_on_globals_closure (display_wayland, &closure->base);
        }
    }
```

**Comment by SirCmpwn** [comment]
Wow, that's stupid.

**Comment by nyorain** [comment]
I will implement it, ok?
Dealt with it quite a bit when working on wlc.

**Comment by acrisci** [comment]
👍

**Comment by acrisci** [comment]
fixed by #107
